### PR TITLE
Fix check for missing/unsupported language

### LINKED ISSUE #3931: Loading Microsoft/CodeContracts with MsBuildWorkspace without BasicWorkspace will throw NRE
Repro:

Clone https://github.com/Microsoft/CodeContracts
Write an application that uses MsBuildWorkspace to load CodeContracts.sln. Make sure to _not_ add BasicWorkspace as a reference
OpenSolutionAsync will throw a NRW because in [MSBuildWorkspace.cs](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Workspace/MSBuild/MSBuildWorkspace.cs,664) commandLineParser is null.

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #3931

A change we made prior to RTM of 1.0 pushed the MSBuild project loading logic into the same dll (workspaces desktop).  However, the existence of the language specific loader service was being used to determine whether the language was supported, which it is not if the language specific workspace library is not present.  This caused unexpected null ref failures later in solution/project loading.

This has been fixed to check for the existence of the command line parser service, as it is still defined in the workspace libraries and is required to succeed at loading.

Added tests to check behavior when workspace language libraries are missing.

@Pilchie @jaredpar @jasonmalinowski @dpoeschl @rchande @CyrusNajmabadi @DustinCampbell  please review

### CODE REVIEW

**Comment by DustinCampbell** [comment]
:+1:

File: src/Workspaces/CoreTest/WorkspaceTests/MSBuildWorkspaceTests.cs [hunk 1]
```diff
@@ -1031,6 +1033,64 @@ public void TestOpenSolution_WithUnrecognizedProjectTypeGuidAndUnrecognizedExten
             });
         }
 
+        private HostServices hostServicesWithoutCSharp = MefHostServices.Create(MefHostServices.DefaultAssemblies.Where(a => !a.FullName.Contains("CSharp")));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
```
Comment (dpoeschl) [inline src/Workspaces/CoreTest/WorkspaceTests/MSBuildWorkspaceTests.cs]: [WorkItem(3931, "https://github.com/dotnet/roslyn/issues/3931"]?

**Comment by dpoeschl** [comment]
:+1:

**Comment by mattwar** [comment]
@dotnet-bot test this please

**Pull request merged**